### PR TITLE
feat(components): setup project and add login

### PR DESCRIPTION
**Pull request by softwarrior**
setup project ans add login

META DATA: issue14

### CODE REVIEW

File: src/features/diary/login/ui/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import styles from './Login.module.scss';
+import { Button, Checkbox, Form, Input } from 'antd';
+import { routes } from 'src/shared/routes';
+
+export const Login = () => {
+  const onFinish = (values: any) => {
+    console.log('Success:', values);
+    routes.diary.open();
```
Comment (Muslik) [inline src/features/diary/login/ui/Login.tsx]: По хорошему роутинг должен быть не ниже pages. То есть pages, processes (у вас скорее не будет использоваться) и app.
widgets, features, entities не знают обычно про какие то редиректы

То есть features, pages по идее вообще не должны знать куда ты будешь редиректить. 

То есть LoginForm может принимать props onSuccess и по нему на странице авторизации уже будешь редиректить

File: src/features/diary/login/ui/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  const onFinishFailed = (errorInfo: any) => {
```
Comment (Muslik) [inline src/features/diary/login/ui/Login.tsx]: onSuccess, onFail

File: src/features/diary/login/ui/Login.tsx (hunk 1 above)
Comment (softwarrior) [reply src/features/diary/login/ui/Login.tsx]: Большое спасибо за ревью. Вынес на уровень выше.

File: src/features/diary/login/ui/Login.tsx (hunk 2 above)
Comment (softwarrior) [reply src/features/diary/login/ui/Login.tsx]: Большое спасибо за ревью. Переименовал.

**Comment by softwarrior** [comment]
Переименовал фичу в auth\loginByPassword.

Исправил замечания.

File: index.html [hunk 3]
```diff
@@ -4,10 +4,16 @@
...
+    <title>JS School Diary</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@400;700&display=swap"
+      rel="stylesheet"
+    />
   </head>
-  <body>
+  <body class="dark">
```
Comment (makhnatkin) [inline index.html]: don't use class attr
all components from antD we will use as is

File: index.html [hunk 4]
```diff
@@ -4,10 +4,16 @@
...
     <div id="root"></div>
-    <script type="module" src="/src/main.tsx"></script>
+    <script type="module" src="/src/index.ts"></script>
```
Comment (makhnatkin) [inline index.html]: there is no need for this file

File: package.json [hunk 5]
```diff
@@ -6,28 +6,41 @@
...
     "@ant-design/icons": "^4.8.0",
     "antd": "^5.0.2",
     "effector": "^22.4.0",
     "effector-react": "^22.3.4",
+    "effector-forms": "^0.0.24",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "history": "^5.3.0",
+    "lodash": "^4.17.21",
```
Comment (makhnatkin) [inline package.json]: there is no point in using Lodash, everything can be done using es5 or another architecture

File: package.json [hunk 6]
```diff
@@ -6,28 +6,41 @@
...
+    "patronum": "^1.12.3"
   },
   "devDependencies": {
     "commitizen": "^4.2.5",
     "@commitlint/cli": "^17.3.0",
+    "@types/lodash": "^4.14.189",
```
Comment (makhnatkin) [inline package.json]: there is no point in using Lodash, everything can be done using es5 or another architecture

File: package.json [hunk 7]
```diff
@@ -6,28 +6,41 @@
...
+    "@types/node": "^18.11.9",
     "@types/react": "^18.0.24",
     "@types/react-dom": "^18.0.8",
     "@vitejs/plugin-react": "^2.2.0",
     "cz-customizable": "^7.0.0",
+    "lint-staged": "^13.0.3",
+    "eslint": "^8.28.0",
+    "prettier": "^2.7.1",
+    "sass": "^1.56.1",
```
Comment (makhnatkin) [inline package.json]: sass is not needed, we will use css in js

File: src/app/index.css [hunk 8]
```diff
@@ -0,0 +1,35 @@
+html,
```
Comment (makhnatkin) [inline src/app/index.css:1]: antd 5 don't use css

File: src/features/diary/login/ui/Login.module.scss [hunk 9]
```diff
@@ -0,0 +1,5 @@
+._ {
```
Comment (makhnatkin) [inline src/features/diary/login/ui/Login.module.scss]: there is no need to use style files, use markup from ant d

File: src/features/diary/login/ui/Login.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
+import styles from './Login.module.scss';
+import { Button, Checkbox, Form, Input } from 'antd';
+import { routes } from 'src/shared/routes';
+
+export const Login = () => {
+  const onFinish = (values: any) => {
```
Comment (makhnatkin) [inline src/features/diary/login/ui/Login.tsx]: do not use any, all types must be described, and if any is needed, then add a comment why any is needed

File: src/features/theme-switcher/ui/ThemeSwitcher.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import { useMobile } from 'src/shared/lib';
+import { Dropdown, DropdownItem } from 'src/shared/ui';
```
Comment (makhnatkin) [inline src/features/theme-switcher/ui/ThemeSwitcher.tsx]: we use only antd components

File: .lintstagedrc [hunk 12]
```diff
@@ -0,0 +1,6 @@
+{
+  "*.{ts,tsx}": ["prettier --write --config .prettierrc"],
+  "{*.json,.huskyrc,.prettierrc,.lintstagedrc,.eslintrc}": [
+    "prettier --write --parser json --config .prettierrc"
+  ]
+}
```
Comment (makhnatkin) [inline .lintstagedrc:6]: this should not be in this pr
use https://github.com/dpolevodin/js_school_diary/pull/11

File: index.html (hunk 3 above)
Comment (softwarrior) [reply index.html]: Thank you. fixed

File: index.html (hunk 4 above)
Comment (softwarrior) [reply index.html]: Thank you. fixed

File: package.json (hunk 5 above)
Comment (softwarrior) [reply package.json]: Thank you. deleted

File: package.json (hunk 6 above)
Comment (softwarrior) [reply package.json]: Thank you. deleted

File: package.json (hunk 7 above)
Comment (softwarrior) [reply package.json]: Thank you. deleted

File: src/features/theme-switcher/ui/ThemeSwitcher.tsx (hunk 11 above)
Comment (softwarrior) [reply src/features/theme-switcher/ui/ThemeSwitcher.tsx]: Thank you. deleted

File: .lintstagedrc (hunk 12 above)
Comment (softwarrior) [reply .lintstagedrc:6]: If they fix it, I'll remove it.

File: src/features/diary/login/ui/Login.module.scss (hunk 9 above)
Comment (softwarrior) [reply src/features/diary/login/ui/Login.module.scss]: use for divs

File: src/app/index.css (hunk 8 above)
Comment (softwarrior) [reply src/app/index.css:1]: use for divs

File: src/features/diary/login/ui/Login.module.scss (hunk 9 above)
Comment (makhnatkin) [reply src/features/diary/login/ui/Login.module.scss]: there is no need to use

File: .lintstagedrc (hunk 12 above)
Comment (makhnatkin) [reply .lintstagedrc:6]: any way this should not be in other pr (like feat(lint))

File: index.html [hunk 13]
```diff
@@ -4,10 +4,16 @@
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <title>Vite + React + TS</title>
+    <title>JS School Diary</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
```
Comment (makhnatkin) [inline index.html:8]: no need other fonts

File: package.json [hunk 14]
```diff
@@ -6,28 +6,38 @@
   "scripts": {
     "dev": "vite",
     "build": "tsc && vite build",
-    "preview": "vite preview",
-    "prepare": "husky install"
+    "preview": "vite preview"
   },
   "dependencies": {
+    "atomic-router": "^0.8.0",
+    "atomic-router-react": "^0.8.0",
```
Comment (makhnatkin) [inline package.json:13]: atomic-router-react in other pr

**Comment by makhnatkin** [comment]
pr is outdated

**Pull request closed without merging**